hasktorch/src/Torch/Typed/Functional.hs

```diff
@@ -2455,23 +2456,9 @@ linear weight bias input = unsafePerformIO $ ATen.cast3 ATen.Managed.linear_ttt
 -- >>> dtype &&& shape &&& (\t' -> D.asValue (toDynamic t') :: [[[[Float]]]]) $ t'
 -- (Float,([1,2,5,2],[[[[0.5,-2.25],[-0.25,1.25],[-2.0,0.0],[0.0,0.5],[1.5,2.5]],[[0.5,-2.25],[-0.25,1.25],[-2.0,0.0],[0.0,0.5],[1.5,2.5]]]]))
 linear'
-  :: forall (inputFeatures :: Nat) (outputFeatures :: Nat) (shape :: [Nat]) (shape' :: [Nat]) dtype device
-   . (shape' ~ CheckBroadcast
-                 (CheckMatMul
-                     shape
-                     '[inputFeatures, outputFeatures]
-                     (ComputeMatMul
-                       (ReverseImpl shape '[]) '[outputFeatures, inputFeatures]))
-                 '[outputFeatures]
-                 (ComputeBroadcast
-                     (ReverseImpl
-                       (CheckMatMul
-                           shape
-                           '[inputFeatures, outputFeatures]
-                           (ComputeMatMul
-                             (ReverseImpl shape '[]) '[outputFeatures, inputFeatures]))
-                       '[])
-                     '[outputFeatures])
+  :: forall (inputFeatures :: Nat) (outputFeatures :: Nat) (shape :: [Nat]) (shape' :: [Nat]) dtype device (shape'' :: [Nat])
+   . ( shape'' ~ MatMul shape '[inputFeatures, outputFeatures]
+     , shape' ~ Broadcast shape'' shape''
```

Comment:
This is much simpler, whyis it the previous version have these check operations interleaved and this one doesnt?

Reply:
the bounds of the previous version were just copied from ghci, I think I used `:type linear'` and copy&pasted that in here. The new bounds are equivalent. If you look up the definition of `MatMul` and `Broadcast`, you'll see that they'll become identical to the previous ones if you expand them.